Common/Product/SharedProject/CommonUtils.cs

```diff
@@ -200,6 +200,26 @@ public static string GetAbsoluteFilePath(string root, string relativePath)
             return string.Join(Path.DirectorySeparatorChar.ToString(), segments);
         }
 
+        /// <summary>
+        /// Returns a normalized file path created by joining relativePath to root.
+        /// The result is not guaranteed to end with a backslash.
+        /// </summary>
+        /// <returns>True, if the absolut path is returned successfully; otherwise false, 
+        /// if root is not an absolute path, or either path is invalid.</returns>
+        public static bool TryGetAbsoluteFilePath(string root, string relativePath, out string absoluteFilePath)
+        {
+            try
+            {
+                absoluteFilePath = GetAbsoluteFilePath(root, relativePath);
+                return true;
+            }
+            catch
```

Comment:
>catch [](start = 12, length = 5)

Do we want to catch exceptions other than `ArgumentException`?

Reply:
So far we only use it for `ArgumentException` as I have not seen it in another scenario. Nonetheless, there are multiple reasons that can make the path fail with different exceptions. These are just a couple of examples :
```
System.Security.SecurityException:
The caller does not have the required permissions.

System.ArgumentNullException:
path is null.

System.NotSupportedException:
path contains a colon (":") that is not part of a volume identifier (for example,
"c:\").

System.IO.PathTooLongException:
The specified path, file name, or both exceed the system-defined maximum length.
For example, on Windows-based platforms, paths must be less than 248 characters,
and file names must be less than 260 characters.
```

I find it's easier and more readable to just catch any exception and return `false`.

Let me know if you think otherwise.

Reply:
Changed as per our offline comments.